src/node/serve/serve.ts

```diff
@@ -31,9 +31,17 @@ export async function serve(options: ServeOptions = {}) {
   const notAnAsset = (pathname: string) => !pathname.includes('/assets/')
   const notFound = fs.readFileSync(path.resolve(site.outDir, './404.html'))
   const onNoMatch: IOptions['onNoMatch'] = (req, res) => {
-    res.statusCode = 404
-    if (notAnAsset(req.path)) res.write(notFound.toString())
-    res.end()
+    if (site.site.entry !== 'index' && req.path === '/') {
```

Comment:
Yeah but this logic won't work when someone is deploying their site. It only works with our custom server which is only suitable for previewing the website locally after build. The servers specifically need `index.html` to be present. That being said, we need to build `entry` as if it was `index.md` (and emit `index.html`).

VuePress has probably done this using their permalink feature, which is still missing in VitePress.

Reply:
I agree, I was worried by this too. I will look into VuePress implementation for this.